app/src/main/java/com/cmm/wdflowers/ui/base/BaseViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.cmm.wdflowers.ui.base
+
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.cmm.wdflowers.R
+
+open class BaseViewModel : ViewModel() {
+
+    // Observable for data content
+    val contentVisibility = MutableLiveData<Int>().apply { View.GONE }
+
+    // Observables for empty state
+    val emptyStateVisibility = MutableLiveData<Int>().apply { View.GONE }
+    val emptyStateMessage = MutableLiveData(R.string.orders_empty_state)
+
+    // Observables for loading view
+    val loadingStateVisibility = MutableLiveData<Int>().apply { View.GONE }
+
+    protected fun setLoadingState() {
+        contentVisibility.postValue(View.GONE)
+        loadingStateVisibility.postValue(View.VISIBLE)
+        emptyStateVisibility.postValue(View.GONE)
+    }
+
+    protected fun setContentState() {
+        contentVisibility.postValue(View.VISIBLE)
+        loadingStateVisibility.postValue(View.GONE)
+        emptyStateVisibility.postValue(View.GONE)
+    }
+
+    protected fun setEmptyState() {
+        contentVisibility.postValue(View.GONE)
+        loadingStateVisibility.postValue(View.GONE)
+        emptyStateVisibility.postValue(View.VISIBLE)
+        R.string.orders_empty_state
```

Comment:
Well, since in this case I've used the same LiveData for the empty/error message, here should have been 
emptyStateMessage.postValue(R.string.orders_empty_state).